zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -366,6 +499,15 @@ impl UpdateWith<ContextuallyValidBlock> for Chain {
             self.revert_chain_state_with(sapling_shielded_data_shared_anchor);
             self.revert_chain_state_with(orchard_shielded_data);
         }
+        if let Some(anchor) = &self.sprout_anchors_by_height.remove(&height) {
+            self.sprout_anchors.remove(anchor);
+        }
+        if let Some(anchor) = &self.sapling_anchors_by_height.remove(&height) {
+            self.sapling_anchors.remove(anchor);
+        }
```

Comment:
What happens if we try to remove an anchor that isn't actually in the multiset?
Should we be asserting something about the return value of `anchors.remove(anchor)`?